src/stan/lang/ast/sigs/function_signatures_def.hpp

```diff
@@ -224,6 +224,32 @@ namespace stan {
           expr_type(double_type()));
     }
 
+    template<typename T>
+    expr_type function_signatures::rng_return_type(const expr_type& t) {
+      T return_type;
+      return t.is_primitive() ?
```

Comment:
You can move the test down and just use its value (which will be 0 or 1).

```
return exp_type(return_type, !t.is_primitive());
```

If you think it's clearer, it could also be `t.is_primitive() ? 0 : 1`.